planex-fetch is confused by local patch sources

/usr/bin/planex-fetch: Failed to fetch oclock-1-cc-headers: Could not resolve host: oclock-1-cc-headers
Makefile:17: recipe for target 'SOURCES/oclock-0.3.tar.gz' failed

-bash-4.3$ grep ^Patch SPECS/ocaml-oclock.spec 
Patch0:         oclock-1-cc-headers
Patch1:         oclock-2-destdir

--- a/planex/fetch.py
+++ b/planex/fetch.py
@@ -5,15 +5,16 @@ planex-fetch: Download sources referred to by a spec file
 import argparse
 import os
 import planex.spec
+import urlparse
 import pycurl
 import sys
 import logging
 import pkg_resources
 
 
-def curl_get(url_string, out_file):
+def curl_get(url, out_file):
     """
-    Fetch the contents of url_string and store to file represented by out_file
+    Fetch the contents of url and store to file represented by out_file
     """
     curl = pycurl.Curl()
 
@@ -33,6 +34,7 @@ def curl_get(url_string, out_file):
     # If we use threads, we should also set NOSIGNAL and ignore SIGPIPE
 
     # Set URL to fetch and file to which to write the response
+    url_string = urlparse.urlunparse(url)
     curl.setopt(pycurl.URL, url_string)
     curl.setopt(pycurl.WRITEDATA, out_file)
 
@@ -80,7 +82,7 @@ def url_for_source(spec, source):
 
     for path, url in zip(spec.source_paths(), spec.source_urls()):
         if path.endswith(source_basename):
-            return url
+            return urlparse.urlparse(url)
 
     raise KeyError(source_basename)
 
@@ -111,20 +113,22 @@ def main(argv):
 
     try:
         url = url_for_source(args.spec, args.source)
-        fetch_http(url, args.source, args.retries + 1)
-
-    except pycurl.error as exn:
-        # Curl download failed
-        sys.exit("%s: Failed to fetch %s: %s" %
-                 (sys.argv[0], url, exn.args[1]))
-
     except KeyError as exn:
         # Source file doesn't exist in the spec
         sys.exit("%s: No source corresponding to %s" % (sys.argv[0], exn))
 
-    except IOError as exn:
-        # IO error saving source file
-        sys.exit("%s: %s: %s" % (sys.argv[0], exn.strerror, exn.filename))
+    if url.scheme in ["http", "https", "file"]:
+        try:
+            fetch_http(url, args.source, args.retries + 1)
+        except pycurl.error as exn:
+            # Curl download failed
+            sys.exit("%s: Failed to fetch %s: %s" %
+                     (sys.argv[0], url, exn.args[1]))
+        except IOError as exn:
+            # IO error saving source file
+            sys.exit("%s: %s: %s" % (sys.argv[0], exn.strerror, exn.filename))
+    else:
+        sys.exit("%s: Unimplemented protocol: %s" % (sys.argv[0], url.scheme))
 
 
 def _main():

--- a/planex/fetch.py
+++ b/planex/fetch.py
@@ -127,6 +127,15 @@ def main(argv):
         except IOError as exn:
             # IO error saving source file
             sys.exit("%s: %s: %s" % (sys.argv[0], exn.strerror, exn.filename))
+
+    elif url.scheme == '' and os.path.dirname(url.path) == '' and \
+            os.path.exists(args.source):
+        # Source file is pre-populated in the SOURCES directory (part of the
+        # repository - probably a patch or local include).   Update its
+        # timestamp to placate make, but don't try to download it.
+        logging.info("Refreshing timestamp for local source %s", args.source)
+        os.utime(args.source, None)
+
     else:
         sys.exit("%s: Unimplemented protocol: %s" % (sys.argv[0], url.scheme))
 

--- a/planex/fetch.py
+++ b/planex/fetch.py
@@ -21,7 +21,7 @@ def curl_get(url, out_file):
     # General options
     useragent = "planex-fetch/%s" % pkg_resources.require("planex")[0].version
     curl.setopt(pycurl.USERAGENT, useragent)
-    curl.setopt(pycurl.FOLLOWLOCATION, 1)
+    curl.setopt(pycurl.FOLLOWLOCATION, True)
     curl.setopt(pycurl.MAXREDIRS, 5)
     curl.setopt(pycurl.CONNECTTIMEOUT, 30)
     curl.setopt(pycurl.TIMEOUT, 300)

--- a/planex/fetch.py
+++ b/planex/fetch.py
@@ -12,7 +12,7 @@ import logging
 import pkg_resources
 
 
-def curl_get(url, out_file):
+def curl_get(url_string, out_file):
     """
     Fetch the contents of url and store to file represented by out_file
     """
@@ -34,7 +34,6 @@ def curl_get(url, out_file):
     # If we use threads, we should also set NOSIGNAL and ignore SIGPIPE
 
     # Set URL to fetch and file to which to write the response
-    url_string = urlparse.urlunparse(url)
     curl.setopt(pycurl.URL, url_string)
     curl.setopt(pycurl.WRITEDATA, out_file)
 
@@ -60,11 +59,12 @@ def fetch_http(url, filename, retries):
     while True:
         retries -= 1
         try:
-            logging.info("Fetching %s to %s", url, filename)
+            url_string = urlparse.urlunparse(url)
+            logging.info("Fetching %s to %s", url_string, filename)
 
             make_dir(os.path.dirname(filename))
             with open(filename, "wb") as out_file:
-                curl_get(url, out_file)
+                curl_get(url_string, out_file)
                 return
 
         except pycurl.error as exn:

